test/e2e/apimachinery/flowcontrol.go

```diff
@@ -317,6 +335,58 @@ func createFlowSchema(f *framework.Framework, flowSchemaName string, matchingPre
 	}
 }
 
+func getFlowSchema(f *framework.Framework, name string) *flowcontrol.FlowSchema {
+	fs, err := f.ClientSet.FlowcontrolV1beta1().FlowSchemas().Get(context.TODO(), name, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	return fs
+}
+
+func getPriorityLevelConfig(f *framework.Framework, name string) *flowcontrol.PriorityLevelConfiguration {
+	pl, err := f.ClientSet.FlowcontrolV1beta1().PriorityLevelConfigurations().Get(context.TODO(), name, metav1.GetOptions{})
+	framework.ExpectNoError(err)
+	return pl
+}
+
+// waitForSteadyState repeatedly polls the API server to check if the newly
+// created flow schema and priority level have been seen by the APF controller
+// by checking: (1) the dangling priority level reference condition in the flow
+// schema status, and (2) metrics. The function times out after 30 seconds.
+func waitForSteadyState(f *framework.Framework, flowSchemaName string, priorityLevelName string) {
+	framework.ExpectNoError(wait.Poll(time.Second, 30*time.Second, func() (bool, error) {
+		fs := getFlowSchema(f, flowSchemaName)
+		if apihelpers.GetFlowSchemaConditionByType(fs, flowcontrol.FlowSchemaConditionDangling) != nil {
+			return false, nil
+		}
+		resp, err := f.ClientSet.CoreV1().RESTClient().Get().RequestURI("/metrics").DoRaw(context.TODO())
+		if err != nil {
+			return false, err
+		}
+		sampleDecoder := expfmt.SampleDecoder{
+			Dec:  expfmt.NewDecoder(bytes.NewBuffer(resp), expfmt.FmtText),
+			Opts: &expfmt.DecodeOptions{},
+		}
+		for {
+			var v model.Vector
+			err := sampleDecoder.Decode(&v)
+			if err != nil {
+				if err == io.EOF {
+					break
+				}
+				return false, err
+			}
+			for _, metric := range v {
+				if string(metric.Metric[model.MetricNameLabel]) != currentExecutingRequestsMetricName {
+					continue
+				}
+				if string(metric.Metric[priorityLevelLabelName]) == priorityLevelName && string(metric.Metric[flowSchemaLabelName]) == flowSchemaName {
```

Comment:
This is not the time series to look for, because this time series is not created until some requests have been classified to this priority level and begun executing.  Instead look at the corresponding time series of the `apiserver_flowcontrol_request_concurrency_limit` metric, that one gets created when the config consuming controller processes the PriorityLevelConfiguration object.

Reply:
@MikeSpreitzer but that doesn't reflect the connection between the flowschema and prioritylevel, by the shared concurrency we don't know whether they're bonded.

Reply:
If this func waits until the FlowSchema has been processed and the PriorityLevelConfiguration has been processed then it is guaranteed (by the current implementation) that the two are connected.  We could remove that reliance on that implementation detail from this test by making it wait until the FlowSchemaStatus says the dandling condition's value is false.

Reply:
Done: re-used the getPLConcurrency and also added a check to make sure the condition is true.